src/Benchmarks/Controllers/HomeController.cs

```diff
@@ -18,12 +19,19 @@ public IActionResult Plaintext()
         }
 
         [HttpGet("json")]
-        [Produces("application/json")]
+        [Produces("application/json", Type = typeof(JsonOutputFormatter))]
         public object Json()
         {
             return new { message = "Hello, World!" };
         }
-        
+
+        [HttpGet("jil")]
+        [Produces("application/json", Type = typeof(JilOutputFormatter))]
```

Comment:
This doesn't do what you want. The `Type` is there for API Explorer

My suggestion:
1. DI MVC options
2. Get the formatters from options
3. `return new ObjectResult(....) { Formatters { myformatter } };`

Reply:
Hmmm, little concerned how much farther away from "idiomatic" this takes us. Is there not a way to do this outside of the controller?

Reply:
You can use a filter, but you need to do the same dance. The JSON formatter isn't trivially constructable.

It's not really *idomatic* to use two separate JSON formatters in an app. We originally had a feature like this but we cut it for simplification purposes.

Reply:
Fair point, hmmm. Perhaps I'll make it replace the JSON formatter instead.